app/routes/project/messages/new.js

```diff
@@ -0,0 +1,66 @@
+import { get } from '@ember/object';
+import { inject as service } from '@ember/service';
+import Route from '@ember/routing/route';
+import AuthenticatedRouteMixin from 'ember-simple-auth/mixins/authenticated-route-mixin';
+import { CanMixin } from 'ember-can';
+import { hash } from 'rsvp';
+
+export default Route.extend(AuthenticatedRouteMixin, CanMixin, {
+  currentUser: service(),
+  store: service(),
+
+  queryParams: { userId: { as: 'user', refreshModel: true } },
+
+  async model({ userId }) {
+    let store = get(this, 'store');
+    let project = this.modelFor('project');
+
+    let { user, currentUser } = await hash({
+      user: store.find('user', userId),
+      currentUser: get(this, 'currentUser.user')
+    });
+
+    let message =  store.createRecord('message', {
+      author: currentUser,
+      initiatedBy: 'admin',
+      project
+    });
+
+    let conversation = store.createRecord('conversation', { user });
+    get(message, 'conversations').pushObject(conversation);
+    return message;
+  },
+
+  actions: {
+    save(message) {
+      // Since it's created without an ID, the store will not detect the
+      // association was saved and will keep it around.
+      // We need to store a reference to it, so we can unload it on save
+      let unsavedConversation = get(message, 'conversations.firstObject');
```

Comment:
Comment explains it. Since we store save the two records (message and conversation) together as one, ember does not detect the conversation child has changed, so we need to unload it manually so it gets replaced with the saved record.